crates/templates/src/source.rs

```diff
@@ -192,6 +203,84 @@ async fn check_local(path: &Path) -> anyhow::Result<LocalTemplateSource> {
     }
 }
 
+async fn download_untar_local(url: &Url) -> anyhow::Result<LocalTemplateSource> {
```

Comment:
Would this make sense as a function on `LocalTemplateSource`? Like `LocalTemplateSource::from_remote_tarball`? Then you can leave a doc comment outlining that it downloads the tarball to a temp dir.

Reply:
The other "pull down a remote source" functions are free, so I've followed the same pattern for now.  We can change them all to methods in a separate refactoring pass if we prefer that.  In the meantime I've added the doc comment.